File: java/byps/src/com/wilutions/byps/BSerializer.java
package com.wilutions.byps;



public abstract class BSerializer {
	
	public final int typeId;	
	
	/**
	 * Store objects without reference and type information.
	 * The object members are directly written into the output buffer.
	 * This member is true, if the class for which this BSerializer
	 * is created is tagged with @inline.
	 */
	public final boolean inlineInstance;
	
	protected BSerializer(int typeId, boolean inlineInstance) {
		this.typeId = typeId;
		this.inlineInstance = inlineInstance;
	}
	
	protected BSerializer(int typeId) {
		this.typeId = typeId;
		this.inlineInstance = false;
	}
	
	/**
	 * Größe des Objekts.
	 * Wenn das Objekt keine Pointer enhält, kann 0 zurückgegeben werden. (String)
	 * @param bout
	 * @return
	 * @throws BException
	 */
	public int size(final Object obj1, final BBinaryModel bmodel) throws BException {
		return 0;
	}
	
	public void write(final Object obj1, final BOutput bout1, final int version) throws BException {
	}
			
	/**
	 * Create and read an object.
	 */
	public abstract Object read(final Object obj1, final BInput bin1, final int version) throws BException;
	
}

File: csharp/byps/BSerializer.cs
/* USE THIS FILE ACCORDING TO THE COPYRIGHT RULES IN LICENSE.TXT WHICH IS PART OF THE SOURCE CODE PACKAGE */
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace byps
{
    public abstract class BSerializer
    {
        public BSerializer(int typeId, bool inlineInstance)
        {
            this.typeId = typeId;
            this.inlineInstance = inlineInstance;
        }

        public BSerializer(int typeId)
        {
            this.typeId = typeId;
            this.inlineInstance = false;
        }

        public virtual int size(Object obj1, BBinaryModel bmodel)
        {
            return 0;
        }

        public virtual void write(Object obj1, BOutput bout1, long version)
        {
        }

        public abstract Object read(Object obj1, BInput bin1, long version);

        public readonly int typeId;
        public readonly bool inlineInstance;
    }
}
